pkgs/tools/misc/qarma-for-qt6/default.qmake.nix

```diff
@@ -0,0 +1,139 @@
+
+/nix/store/4gllgq492npdmvwc6gb5kvi119libj21-qtbase-6.2.1-dev/mkspecs/features/qt_config.prf
+         QT_MODULE_INCLUDE_BASE = $$[QT_INSTALL_HEADERS]
+         QT_MODULE_LIB_BASE = $$[QT_INSTALL_LIBS]
+         QT_MODULE_HOST_LIB_BASE = $$[QT_HOST_LIBS]
+         QT_MODULE_HOST_LIBEXEC_BASE = $$[QT_HOST_LIBEXECS]
+         QT_MODULE_BIN_BASE = $$[QT_INSTALL_BINS]
+
+
+*/
+
+mkDerivation rec {
+  pname = "qarma";
+  version = "2021-10-05";
+
+  src = fetchFromGitHub {
+    name = "${pname}-${version}-source";
+    owner = "luebking";
+    repo = pname;
+    rev = "605ea4213406718ba869dd146875195e57488786";
+    sha256 = "KFoFywFeGqNmE1y49DrXJZ1jIK5jMOCOspkkFME+DR8=";
+  };
+
+  # FIXME only needed for qt6
+  # set attributes for qt-6/hooks/qmake-hook.sh
+  # TODO better. we do not want to set this for every libsForQt6.callPackage target
+  inherit (qtbase) qtDocPrefix qtQmlPrefix qtPluginPrefix;
+
+  patches = [
+    ./qarma-qt6.patch
+  ];
+
+  patchPhase = ''
+
+    echo "debug: qtbase.out = ${qtbase.out}"
+    echo "debug: qtbase.dev = ${qtbase.dev}"
+    echo "debug: qtbase.bin = ${qtbase.bin}"
+
+    sed -i.bak -E -e "s,(target\.path \+=) /usr/bin,\1 $out/bin," -e 's/x11extras/core-private/g' qarma.pro
+
+    # qtbase-6.2.1-dev/mkspecs/linux-g++
+    export QMAKEPATH="${qtbase.out}:${qtbase.dev}:$QMAKEPATH"
+
+    #export LD_LIBRARY_PATH="${qtbase.out}/lib:$LD_LIBRARY_PATH"
+
+    # /nix/store/i9vknhf60qi16k64h5zspmq6ysiijym4-qtbase-6.2.1-dev/mkspecs/common/linux.conf
+    export QMAKE_LIBDIR="${qtbase.out}/lib:$QMAKE_LIBDIR"
+    export QMAKE_INCDIR="${qtbase.dev}/include:$QMAKE_INCDIR"
+
+    # workaround for: fatal error: QApplication: No such file or directory
+    # no effect: export QMAKE_INCDIR="${qtbase.dev}/include/QtWidgets:$QMAKE_INCDIR"
+    # probably we need to patch qmake files, so qmake can find headers
+
+    # workaround for bug in qtbase.nix
+    export QMAKE_INCDIR_OPENGL_ES2="${libglvnd.dev}/include"
+    export QMAKE_LIBDIR_OPENGL_ES2="${libglvnd.out}/lib"
+
+    # workaround for bug in qtbase-6.2.1-dev/mkspecs/features/qt_config.prf
+    # no effect: export QMAKE_QT_CONFIG="${qtbase.dev}/mkspecs/qconfig.pri"
+
+    # no effect: export QT_INSTALL_HEADERS="${qtbase.dev}/include"
+
+    echo "debug: LD_LIBRARY_PATH = $LD_LIBRARY_PATH"
+  '';
+
+  # https://doc.qt.io/qt-5/qmake-environment-reference.html
+  preConfigure = ''
+    export HOME=/tmp
+
+    # no effect
+    echo qmake -set QT_INSTALL_HEADERS "${qtbase.dev}/include"
+    qmake -set QT_INSTALL_HEADERS "${qtbase.dev}/include"
+
+    echo 'qmake -query QT_INSTALL_HEADERS'
+    qmake -query QT_INSTALL_HEADERS
+
+    echo 'qmake -query | grep QT_INSTALL_HEADERS'
+    qmake -query | grep QT_INSTALL_HEADERS
+
+    find /tmp/.config
+    cat /tmp/.config/QtProject/QMake.conf
+
+    echo QT_INSTALL_HEADERS
+    echo "expected = ${qtbase.dev}/include"
+    echo "actual   = $(qmake -query QT_INSTALL_HEADERS)"
```

Comment:
qmake is failing here

```
cd nixpkgs
nix-build . -A qarmaForQt6Qmake

expected = /nix/store/26k5lszsa2adc2qxj5yqxc06ygbq0zf3-qtbase-6.2.1-dev/include
actual   = /nix/store/52p7a812jmy3fl63f349dq1mwlai3xir-qtbase-6.2.1/include
```

qmake expects qt headers in qtbase $out, but headers are in qtbase $dev

`qmake2cmake` aka [pro2cmake.py](https://code.qt.io/cgit/qt/qtbase.git/tree/util/cmake/pro2cmake.py) is a workaround to build qmake projects
example project is [qarma](https://github.com/milahu/nixpkgs/blob/qt-6-init/pkgs/tools/misc/qarma-for-qt6/default.nix) &rarr; ```nix-build . -A qarmaForQt6```

Reply:
should be fixed in c5928ecb52c81bb8aeb5fe37591d74b8bf519668

at least `qmake -query QT_INSTALL_HEADERS` is now returning the correct value